core/linux.c

```diff
@@ -83,18 +83,17 @@ const char *system_default_directory(void)
 
 const char *system_default_filename(void)
 {
-	static char *filename = NULL;
-	if (!filename) {
+	static const char *path = NULL;
+	if (!path) {
 		const char *user = getenv("LOGNAME");
 		if (same_string(user, ""))
 			user = "username";
-		filename = calloc(strlen(user) + 5, 1);
+		char *filename = calloc(strlen(user) + 5, 1);
 		strcat(filename, user);
 		strcat(filename, ".xml");
-	}
-	static const char *path = NULL;
-	if (!path)
 		path = system_default_path_append(filename);
+		free(filename);
+	}
```

Comment:
And of course with this change here, the other PR that I closed a moment ago makes more sense as you changed how things are allocated. Would have been nice if the other PR had mentioned this one.

Reply:
Sorry, but you're reading it wrong - these two PRs are perfectly independent. Note that it's *not* "filename" that is returned here, but "path". filename *always* leaked in this function.

If you want, I can combine both commits in a single PR, still you're analysis is flawed.